lib/rbi/rbs/type_translator.rb

```diff
@@ -0,0 +1,164 @@
+          )
+        end
+
+        #: (NodeType) -> Type
+        def translate(type)
+          case type
+          when ::RBS::Types::Alias
+            # TODO: unsupported yet
+            Type.untyped
+          when ::RBS::Types::Bases::Any
+            Type.untyped
+          when ::RBS::Types::Bases::Bool
+            Type.boolean
+          when ::RBS::Types::Bases::Bottom
+            Type.noreturn
+          when ::RBS::Types::Bases::Class
+            # TODO: unsupported yet
+            Type.untyped
+          when ::RBS::Types::Bases::Instance
+            Type.attached_class
+          when ::RBS::Types::Bases::Nil
+            Type.simple("NilClass")
+          when ::RBS::Types::Bases::Self
+            Type.self_type
+          when ::RBS::Types::Bases::Top
+            Type.anything
+          when ::RBS::Types::Bases::Void
+            Type.void
+          when ::RBS::Types::ClassSingleton
+            Type.class_of(Type.simple(translate_name(type.name)))
+          when ::RBS::Types::ClassInstance
+            Type.simple(translate_name(type.name))
+          when ::RBS::Types::Function
+            translate_function(type)
+          when ::RBS::Types::Interface
+            # TODO: unsupported yet
+            Type.untyped
+          when ::RBS::Types::Intersection
+            Type.all(*type.types.map { |t| translate(t) })
+          when ::RBS::Types::Literal
+            # TODO: unsupported yet
+            Type.untyped
+          when ::RBS::Types::Optional
+            Type.nilable(translate(type.type))
+          when ::RBS::Types::Proc
+            proc = T.cast(translate(type.type), Type::Proc)
+            proc.bind(translate(type.self_type)) if type.self_type
+            proc
+          when ::RBS::Types::Record
+            Type.shape(type.fields.map { |name, type| [name, translate(type)] }.to_h)
+          when ::RBS::Types::Tuple
+            Type.tuple(type.types.map { |t| translate(t) })
+          when ::RBS::Types::Union
+            Type.any(*type.types.map { |t| translate(t) })
+          when ::RBS::Types::UntypedFunction
+            Type.proc.params(arg0: Type.untyped).returns(Type.untyped)
+          when ::RBS::Types::Variable
+            Type.type_parameter(type.name)
+          else
+            T.absurd(type)
+          end
+        end
+
+        private
+
+        #: (::RBS::TypeName) -> String
+        def translate_name(name)
+          string = String.new
+
+          namespace = name.namespace
+          string << "::" if namespace.absolute?
+
+          path = namespace.path
+          string << path.join("::")
+          string << "::" if path.any?
+
+          string << name.name.to_s
+
+          string
+        end
+
+        #: (::RBS::Types::Function) -> Type
+        def translate_function(type)
+          proc = Type.proc
+
+          index = 0
+
+          type.required_positionals.each do |param|
+            proc.proc_params[:"arg#{index}"] = translate(param.type)
```

Comment:
These params could have declared names, no? Shouldn't we be using anonymous names only when the names don't exist?

Reply:
Oh yeah, we should keep the names if they exist. Changed the code and added a test 👍